sdcm/tester.py

```diff
@@ -883,70 +884,89 @@ def setUp(self):  # pylint: disable=too-many-branches,too-many-statements
                 if not k8s_cluster.chaos_mesh.initialized:
                     k8s_cluster.chaos_mesh.initialize()
 
-        if self.db_cluster and not self.db_clusters_multitenant:
-            self.db_clusters_multitenant = [self.db_cluster]
-        for db_cluster in self.db_clusters_multitenant:
-            if not (db_cluster and db_cluster.nodes):
-                continue
-            if self.params.get('use_legacy_cluster_init'):
-                self.legacy_init_nodes(db_cluster=db_cluster)
-            else:
-                self.init_nodes(db_cluster=db_cluster)
-
-            if self.params.get('use_ldap'):
-                with temp_authenticator(db_cluster.nodes[0], "org.apache.cassandra.auth.PasswordAuthenticator"):
-                    # running `set_system_auth_rf()` before changing authorization/authentication protocols
+        def _create_db_clusters():
+            if self.db_cluster and not self.db_clusters_multitenant:
+                self.db_clusters_multitenant = [self.db_cluster]
+            for db_cluster in self.db_clusters_multitenant:
+                if not (db_cluster and db_cluster.nodes):
+                    continue
+                if self.params.get('use_legacy_cluster_init'):
+                    self.legacy_init_nodes(db_cluster=db_cluster)
+                else:
+                    self.init_nodes(db_cluster=db_cluster)
+
+                if self.params.get('use_ldap'):
+                    with temp_authenticator(db_cluster.nodes[0], "org.apache.cassandra.auth.PasswordAuthenticator"):
+                        # running `set_system_auth_rf()` before changing authorization/authentication protocols
+                        self.set_system_auth_rf(db_cluster=db_cluster)
+
+                        self._setup_ldap_roles(db_cluster=db_cluster)
+                        if self.params.get('ldap_server_type') == LdapServerType.MS_AD:
+                            change_default_password(node=db_cluster.nodes[0],
+                                                    user=self.params.get('authenticator_user'),
+                                                    password=self.params.get('authenticator_password'))
+                            # TODO: update proper loader and monitor in multi-tenant case.
+                            #       Now it updates only the first ones all the time.
+                            self.loaders.added_password_suffix = True
```

Comment:
This looks racy.
Probably will fail when LDAP setup is enabled.

Reply:
I moved it to separate block after setup db/loaders/monitors in sequential way - should not impact setup time.